pzsd_bot/__main__.py

```diff
@@ -187,6 +191,98 @@ async def leaderboard(ctx):
     await ctx.respond(embed=embed)
 
 
+@bot.slash_command(description="Add new name that can be bestowed points.")
+@option("name", description="The exact name to use when bestowing points.")
+@option("snowflake", description="Their discord ID if applicable.", required=False)
+@default_permissions(administrator=True)
+async def register(ctx, name, snowflake):
+    logger.info(
+        "%s invoked /register with name=%s and snowflake=%s",
+        ctx.author.name,
+        name,
+        snowflake,
+    )
+
+    async with engine.connect() as conn:
+        result = await conn.execute(select(pzsd_user).where(pzsd_user.c.name == name))
+
+    user_to_add = result.one_or_none()
+    if user_to_add is not None:
+        if user_to_add.is_active:
+            logger.info("User '%s' already exists", name)
+            await ctx.respond(f"'{name}' already exists!")
+            return
+        else:
+            logger.info("User '%s' exists but is inactive", name)
+            async with engine.begin() as conn:
+                await conn.execute(
+                    update(pzsd_user)
+                    .where(pzsd_user.c.name == name)
+                    .values(is_active=True, discord_snowflake=snowflake)
+                )
+    else:
+        async with engine.begin() as conn:
+            await conn.execute(
+                insert(pzsd_user).values(
+                    name=name,
+                    discord_snowflake=snowflake,
+                )
+            )
+
+    logger.info("Added user '%s' to user table", name)
```

Comment:
Not true if reactivating user

Reply:
Thought I might get push back on the wording here. I figured I'd just consider disabling a user as actually "deleting" them, since I'm ultimately just trying to avoid actual deletion of data as that would force me to delete rows in the ledger table connected to the user being deleted. So I felt like "added" was fine enough as a generic log to mean both adding a new user and reactivating them.

If you really feel strongly about it, I could make the logs more accurate.